src/main/java/org/intellij/sdk/codesync/database/Database.java

```diff
@@ -17,43 +17,52 @@ public class Database {
 
     private static Connection connection = null;
 
-    public static boolean isConnected(){
-        return connection != null;
+    public static void setConnectionNull() {
+        //Method for unit testing.
+        Database.connection = null;
     }
 
-    public static void initiate() {
-        try{
-            File file = new File(DATABASE_PATH);
-            if(!file.exists()){
-                Class.forName("org.sqlite.JDBC");
-                connection = DriverManager.getConnection(CONNECTION_STRING);
+    public static void setupDbFilesAndTables(String databasePath, String connectionString){
+
+        /*
+            This method will make sure to create db file and required tables,
+            make migration from files to db if necessary.
+        */
+
+        File file = new File(databasePath);
+        if(!file.exists()){
+            try {
+                initiateConnection(connectionString);
```

Comment:
We are doing this (conditionally) inside executeUpdate as well, maybe we can remove from here?

Reply:
How about changing `Database.executeUpdate` to `Database().executeUpdate`? i.e. move iniitateConnection to constructor so it implicility connect/reconnect before every query run? @syedgulahmed

Reply:
Yeah you are right but due to some reasons we can't do that because it will not work. I understand there is lot of code redundancy, we can reduce the code and refactor it by using Singleton design pattern. If we don't do deploy today, I can work on that today and we can have much better code and won't have redundant calls like isConnected, initiateConnection etc.

Reply:
Sounds good to me @syedgulahmed If we can deploy by tomorrow with better code, that's good.